test/contracts/OVM/bridge/base/OVM_L2CrossDomainMessenger.spec.ts

```diff
@@ -157,5 +158,40 @@ describe('OVM_L2CrossDomainMessenger', () => {
         OVM_L2CrossDomainMessenger.relayMessage(target, sender, message, 0)
       ).to.be.revertedWith('Provided message has already been received.')
     })
+
+    it('should not make a call if the target is the L2 MessagePasser', async () => {
+      Mock__OVM_L1MessageSender.smocked.getL1MessageSender.will.return.with(
+        Mock__OVM_L1CrossDomainMessenger.address
+      )
+      target = await AddressManager.getAddress('OVM_L2ToL1MessagePasser')
+      message = Mock__OVM_L2ToL1MessagePasser.interface.encodeFunctionData('passMessageToL1(bytes)', [
+        NON_NULL_BYTES32,
+      ])
+
+      const resProm = OVM_L2CrossDomainMessenger.relayMessage(target, sender, message, 0)
+
+      // The call to relayMessage() should succeed.
+      await expect(
+        resProm
+      ).to.not.be.reverted
+
+      // There should be no 'relayedMessage' event logged in the receipt.
+      const logs = (
+        await Mock__OVM_L2ToL1MessagePasser.provider.getTransactionReceipt(
+          (await resProm).hash
+        )
+      ).logs
+      expect(logs).to.deep.equal([])
```

Comment:
This check is an indirect way of verifying that "no call was made to the Message Passer", I would have preferred to use `Mock__OVM_L2ToL1MessagePasser.smocked.passMessageToL1.calls.length === 0`, but smock doesn't actually handle the case where no calls are made to a function. I spend some time on a fix, but other tests broke, so I went with this.